test/Network/MQTT/ParserSpec.hs

```diff
@@ -354,6 +354,53 @@ spec = do
             , 0x00, 0x01, userName
             ]
 
+      it "Will topic should not contain wildcards" $ do
+        shouldFailParsing
+          [0x10,
+           -- remaining length = 33 bytes
+           33,
+           -- protocol name = MQTT
+           0x00, 0x04, 0x4d, 0x51, 0x54, 0x54,
+           -- protocol level = 4
+           0x04,
+           -- username = 1
+           -- password = 1
+           -- retain   = 0
+           -- willqos  = 01
+           -- will     = 1
+           -- clean    = 1
+           -- reserved = 0
+           0xce,
+           -- keep alie = 0x000c
+           0x00, 0x0c,
+           -- client id = "abc"
+           0x00, 0x03, 0x61, 0x62, 0x63,
+           -- topic name = "a/+"
+           0x00, 0x03, 0x61, 0x2f, fromIntegral $ fromEnum '+']
```

Comment:
Consider changing all occurrences `fromIntegral $ fromEnum` or `fromIntegral . fromEnum` to `toWord8`